ecl/hql/hqlopt.cpp

```diff
@@ -3786,6 +3786,18 @@ IHqlExpression * CTreeOptimizer::doCreateTransformed(IHqlExpression * transforme
             node_operator childOp = child->getOperator();
             switch(childOp)
             {
+            //None of the follow change the format of the left hand side, and since the fetch is on the underlying
+            //dataset they can all be removed.  If they remain the disk file can be treated as if it is in memory format.
+            case no_preload:
+            case no_nofold:
+            case no_nocombine:
+            case no_dedup:
+            case no_group:
+            case no_compound_diskread:
+            case no_filter:
```

Comment:
@ghalliday Is no_dedup and no_filter valid operation on the dataset used in a fetch.  If not shouldn't it throw an exception?

Reply:
Apparently it is - I have test cases which check it is supported.  It probably shouldn't be, but I suspect it was in response to a bug report. (fetch11)